mysql-test/suite/binlog/t/binlog_error_action.test

```diff
@@ -0,0 +1,173 @@
+
+SET GLOBAL binlog_error_action= ABORT_SERVER;
+--exec echo "wait" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--error ER_BINLOG_LOGGING_IMPOSSIBLE
+flush logs;
+--chmod 0666 $INDEX
+--exec echo "restart" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--enable_reconnect
+--source include/wait_until_connected_again.inc
+
+--echo Test case2
+# Make MYSQL_BIN_LOG::open fail with debug simulation and test
+# the behaviour in case of ABORT
+SET SESSION debug="+d,fault_injection_updating_index";
+SET GLOBAL binlog_error_action= ABORT_SERVER;
+--exec echo "wait" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--error ER_BINLOG_LOGGING_IMPOSSIBLE
+flush logs;
+--exec echo "restart" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--enable_reconnect
+--source include/wait_until_connected_again.inc
+
+--echo Test case3
+# Simulate diskfull during opening of binlog and check for diskfull error
+# behaviour where binlog error action is to abort the server.
+SET SESSION debug="+d,simulate_file_write_error";
+SET GLOBAL binlog_error_action= ABORT_SERVER;
+--exec echo "wait" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--error ER_BINLOG_LOGGING_IMPOSSIBLE
+flush logs;
+--exec echo "restart" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--enable_reconnect
+--source include/wait_until_connected_again.inc
+
+--echo Test case4
+SET SESSION debug="+d,fault_injection_init_name";
+SET GLOBAL binlog_error_action= ABORT_SERVER;
+--exec echo "wait" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--error ER_BINLOG_LOGGING_IMPOSSIBLE
+flush logs;
+--exec echo "restart" > $MYSQLTEST_VARDIR/tmp/mysqld.1.expect
+--enable_reconnect
+--source include/wait_until_connected_again.inc
+
+--echo Test case5
+# Make binlog index readonly and check the behaviour with IGNORE
+--chmod 0444 $INDEX
+--replace_regex /\.[\\\/]mysqld/mysqld/ /Errcode: 13 ".*"\)/Errcode: 13 "Permission denied")/
+--error 29
+flush logs;
+CREATE TABLE t1 ( f int );
+SHOW TABLES;
+DROP TABLE t1;
+--chmod 0666 $INDEX
+--source include/restart_mysqld.inc
+
+--echo Test case6
+# Make MYSQL_BIN_LOG::open_binlog fail with debug simulation and test
+# the behaviour in case of IGNORE
+SET SESSION debug="+d,fault_injection_updating_index";
+--replace_regex /\.[\\\/]master/master/ /errno: 1 ".*"\)/errno: 1 "Operation not permitted")/
+--error ER_CANT_OPEN_FILE
+flush logs;
+CREATE TABLE t2 (f int );
+SHOW TABLES;
+DROP TABLE t2;
+SET SESSION debug="-d,fault_injection_updating_index";
+--source include/restart_mysqld.inc
+
+#FIXME: hangs MTR
+#--echo Test case7
+## Simulate diskfull during opening of binlog and check for diskfull error
+## behaviour where binlog error action is to continue the server
+## after the error.
+#SET SESSION debug="+d,simulate_file_write_error";
+#--replace_regex /\.[\\\/]master/master/ /Errcode: 28 ".*"\)/Errcode: 28 "No space left on device")/
+## error EE_WRITE with error code number 3 is expected.
+#--error 3
+#flush logs;
+#SET SESSION debug="-d,simulate_file_write_error";
+#--source include/restart_mysqld.inc
```

Comment:
[MDEV-20796](https://jira.mariadb.org/browse/MDEV-20796) notes that MariaDB would continuously retry writing every minute until it no longer encounters a full disk error. It reminded me that I saw this retry message in my logs; they confirm this “hanging” behavior.

Reply:
Well, in general, we don't have to, but if we _can_ conform to `MySQL` options, it is preferred, as it makes migration more straightforward. Another option would be to include MDEV-20796 in the scope of this PR (perhaps a separate commit though) as a third option that allows for consistency between the option names and behaviors.

Reply:
> Another option would be to include MDEV-20796 in the scope of this PR

Nay, the option MDEV-20796 wants to make other cases continuously retry as well; it’s a distinct feature.

Reply:
Right, but it would allow each option to be consistent to its meaning. That is, we could slightly change `IGNORE_ERROR` behavior so it wouldn't "retry later" on full disk errors, but it would always turn off binlogging and move on. And we wouldn't lose the functionality because it would be nested in the `RETRY` configuration.

Reply:
MySQL `IGNORE_ERROR` seems to actually mean `DISABLE_LOGGING`. 🤨🤔